dali/pipeline/op_graph.h

```diff
@@ -46,35 +57,62 @@ struct OpNode {
   }
 
   std::unique_ptr<OperatorBase> op;
-  NodeID id;
+  OpNodeId id;
   OpSpec spec;
-  std::set<NodeID> parents, children;
+  std::set<OpNodeId> parents, children;
+
+  // parent and children tensors indexed by our inputs and outputs
+  std::vector<TensorNodeId> parent_tensors, children_tensors;
+
+  // parent ops indexed by our inputs,
+  // parent_ops[i] === parent_tensors[i].producer_edge.node;
+  // std::vector<OpNodeId> parent_ops;
+
   std::string instance_name;
+  DALIOpType op_type;
+  OpPartitionId partition_index;
 };
 
 // Stores meta-data about a tensor and how it
 // is used by a producer/consumer node.
 struct TensorMeta {
-  NodeID node;
+  OpNodeId node;
+  TensorNodeId tensor;
```

Comment:
This field seems redundant - it's never read - just written to; it contains the id of the containing TensorNode, so it doesn't seem to be very useful.

Reply:
Done